flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDeserializationSchema.java

```diff
@@ -130,6 +135,9 @@ public Row deserialize(byte[] message) throws IOException {
 			final JsonNode root = objectMapper.readTree(message);
 			return (Row) runtimeConverter.convert(objectMapper, root);
 		} catch (Throwable t) {
+			if (ignoreParseErrors) {
+				return null;
+			}
```

Comment:
It ignores all exceptions including the exception which may be not a parsing errors. For example, if `ignoreParseErrors=true` and `failOnMissingField=true`, the missing field exception will be swallowed. 

I think we should check the exception in every converters to have a fine-grained exception ignoring.

Reply:
When we set ignoreParseErrors=true and failOnMissingField=true, what is the expected behavior if we encounter an invalid json string like "DDD" ?

Reply:
We have two parsing error actually:  
- `objectMapper.readTree` will throw parsing error when data is not a valid json, it will throw `IOException`
- converters will throw parsing error when data cannot converted to specified type, we can make converters throw a custom exception

So we can check exceptions for just these two types of exception, and apply ignore parse errors strategy.